rust-bindings/src/lib.rs

```diff
@@ -0,0 +1,299 @@
+use std::{
+    ffi::{CStr, CString},
+    os::raw::c_char,
+};
+
+use anyhow::{anyhow, Result};
+use concordium_contracts_common::{
+    schema::{Type, VersionedModuleSchema},
+    Cursor,
+};
+use serde_json::to_string;
+
+pub type HexString = String;
+pub type JsonString = String;
+
+#[repr(C)]
+pub struct FFIByteOption {
+    pub t: u8,
+    pub is_some: u8,
+}
+
+impl FFIByteOption {
+    pub fn into_option(self) -> Option<u8> {
+        match self.is_some {
+            1 => Option::Some(self.t),
+            _ => Option::None,
+        }
+    }
+}
+
+/// Get module schema in a human interpretable form.
+///
+/// # Arguments
+///
+/// * 'schema' - Module schema in hexadecimal
+/// * 'schem_version' - Optional schema version
+/// * 'result' - Parsed schema if the call succeeded or the error message in case of failure.
+///
+/// # Returns
+///
+/// If the call succeeded or not.
+///
+/// # Safety
+///
+/// This function is marked as unsafe because it performs operations that are not checked by the
+/// Rust compiler.
```

Comment:
Usually the safety section should explain how to use it safely.

Reply:
Sections updated